### PR TITLE
doc: Adds documentation & examples for `mongodbatlas_encryption_at_rest` singular data source

### PR DESCRIPTION
**Pull request by maastha**
## Description

- Added EAR Azure example
- Added documentation & templates for `mongodbatlas_encryption_at_rest` singular data source
- Updated GHA to check docs

Link to any related issue(s): [CLOUDP-267668](https://jira.mongodb.org/browse/CLOUDP-267668)

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR. A migration guide must be created or updated if the new feature will go in a major version.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR. A migration guide must be created or updated.
- [ ] This change requires a documentation update
- [x] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contributing guides](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/README.md)
- [ ] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [ ] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [ ] I have added any necessary documentation (if appropriate)
- [ ] I have run make fmt and formatted my code
- [ ] If changes include deprecations or removals I have added appropriate changelog entries.
- [ ] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

**Comment by maastha** [comment]
This PR currently depends on merging https://github.com/mongodb/terraform-provider-mongodbatlas/pull/2538

File: docs/resources/encryption_at_rest.md [hunk 1]
```diff
@@ -63,49 +90,50 @@ resource "mongodbatlas_encryption_at_rest" "default" {
...
+data "mongodbatlas_encryption_at_rest" "test" {
+  project_id = mongodbatlas_encryption_at_rest.test.project_id
+}
+
+output "is_azure_encryption_at_rest_valid" {
+  value = data.mongodbatlas_encryption_at_rest.test.azure_key_vault_config.valid
+}
 ```
 
+-> **NOTE:** It is possible to configure Atlas Encryption at Rest to communicate with Azure Key Vault using Azure Private Link, ensuring that all traffic between Atlas and Key Vault takes place over Azure’s private network interfaces. Please review `mongodbatlas_encryption_at_rest_private_endpoint` resource for details.
```
Comment (AgustinBettati) [inline docs/resources/encryption_at_rest.md:124]: good callout

**Review by AgustinBettati — APPROVED** [review]
LGTM

**Review by lantoli — APPROVED** [review]

**Review by EspenAlbert — APPROVED** [review]
LGTM

File: docs/data-sources/encryption_at_rest.md [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+<!-- schema generated by tfplugindocs -->
+## Schema
+
+### Required
+
+- `project_id` (String) Unique 24-hexadecimal digit string that identifies your project. Use the [/groups](#tag/Projects/operation/listProjects) endpoint to retrieve all projects to which the authenticated user has access.
+
+**NOTE**: Groups and projects are synonymous terms. Your group id is the same as your project id. For existing groups, your group/project id remains the same. The resource and corresponding endpoints use the term groups.
+
+### Read-Only
```
Comment (AgustinBettati) [inline docs/data-sources/encryption_at_rest.md]: As spoken offline, would consider doing adjustment done in private_endpoint resource for alignment (https://github.com/mongodb/terraform-provider-mongodbatlas/pull/2547#discussion_r1741680470)

**Review by davidhou17 — APPROVED** [review]

File: docs/data-sources/encryption_at_rest.md (hunk 2 above)
Comment (maastha) [reply docs/data-sources/encryption_at_rest.md]: updated

**Pull request merged**